src/Service.Tests/SqlTests/GraphQLSupportedTypesTests/GraphQLSupportedTypesTestsBase.cs

```diff
@@ -637,35 +662,43 @@ private static void ValidateArrayResults(JsonDocument actualJsonDoc, JsonDocumen
                 actualElement.TryGetProperty(fieldName, out JsonElement actualValue);
                 expectedElement.TryGetProperty(fieldName, out JsonElement expectedValue);
 
-                if (fieldName.StartsWith(DATETIMEOFFSET_TYPE.ToLower()))
-                {
-                    DateTimeOffset actualDateTimeOffset = DateTimeOffset.Parse(actualValue.ToString(), DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal);
-                    DateTimeOffset expectedDateTimeOffset = DateTimeOffset.Parse(expectedValue.ToString(), DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal);
-                    Assert.AreEqual(actualDateTimeOffset.ToString(), expectedDateTimeOffset.ToString());
-                    // Comparing for Miliseconds separetly since Hotcholate respond with only 3 decimal places.
-                    Assert.AreEqual(actualDateTimeOffset.Millisecond, expectedDateTimeOffset.Millisecond);
-                }
-                else if (fieldName.StartsWith(DATETIME_TYPE.ToLower()))
-                {
-                    // MySql returns a format that will not directly parse into DateTime type so we use string here for parsing
-                    DateTime actualDateTime = DateTime.Parse(actualValue.ToString(), CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal);
-                    DateTime expectedDateTime = DateTime.Parse(expectedValue.ToString(), CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal);
-                    Assert.AreEqual(expectedDateTime, actualDateTime);
-                }
-                else if (fieldName.StartsWith(SINGLE_TYPE.ToLower()))
-                {
-                    Assert.AreEqual(expectedValue.GetSingle(), actualValue.GetSingle());
-                }
-                else if (fieldName.StartsWith(TIME_TYPE.ToLower()))
-                {
-                    TimeOnly actualTime = TimeOnly.Parse(actualValue.ToString());
-                    TimeOnly expectedTime = TimeOnly.Parse(expectedValue.ToString());
-                    Assert.AreEqual(expectedTime.ToLongTimeString(), actualTime.ToLongTimeString());
-                }
-                else
-                {
-                    Assert.AreEqual(expectedValue.GetDouble(), actualValue.GetDouble());
-                }
+                AssertOnFields(fieldName, actualValue.ToString(), expectedValue.ToString());
+            }
+        }
+
+        /// <summary>
+        /// Compare given fields from actual and expected json.
+        /// </summary>
+        private static void AssertOnFields(string field, string actualElement, string expectedElement)
+        {
+            if (field.StartsWith(DATETIMEOFFSET_TYPE.ToLower()))
```

Comment:
Does it have to be `StartsWith` or `Equals`?

Reply:
StartsWith because the field here is the name of the column, ex: datetime_types, while the const variable here just represent the type, i.e, datetime.